### ISSUE TITLE
Public Disclosure of issue 726

### ISSUE THREAD

**Issue by Kvarkas** [mRemoteNG/mRemoteNG#2420]
The passwords of the connection files are stored as plain text values in the memory of the mRemteNG process, even before a
connection is established.
This fulfills the requirement of CWE-316 - Cleartext Storage of Sensitive
Information in Memory.
https://github.com/mRemoteNG/mRemoteNG/issues/726

1) Create a new connection configuration, set hostname, username and
password.
2) (Optional) Enable encryption of the configuration file.
3) Restart mRemoteNG
4) Use Task Manager / Processhacker / Procdump or any other tool that can
create minidumps of a process to create a minidump.
5) Examine the minidump file and look for the password from step one.

## Your Environment

* Version used: 1.77.3 dev
* Windows 11

Reported by Maximilian Barz

**Comment by S1lkys** [comment]
CVE-2023-30367 has been published: 

- https://github.com/S1lkys/CVE-2023-30367-mRemoteNG-password-dumper
- https://www.secuvera.de/advisories/secuvera-SA-2023-01.txt

**Comment by Neustradamus** [comment]
To follow this ticket.

**Comment by felsgaertner** [comment]
Hello, as this issue was opened almost one year ago and is marked "high priority", I'd like to ask for the current status.

**Comment by Hillage** [comment]
Hi there, is there an update on this CVE? Is it in the works, or perhaps fixed in the NB with .Net 8 support?
Thanks! 😄

**Comment by Kvarkas** [comment]
It in developing, will be implemented for both versions

**Comment by felsgaertner** [comment]
@kvarkas, do you have a time estimation, when this will be ready for release (at least a rough estimation in months)?

**Comment by Kvarkas** [comment]
expecting this spring, still need to do testing, have a bit limiting free time for coding

**Comment by Schwanzskala** [comment]
Hello @Kvarkas,  is there any news on when this might be ready?

**Comment by Kvarkas** [comment]
working on it, expected end of next week, thanks

**Comment by mi3technosouth** [comment]
@Kvarkas Any news or expected timeline of release?

**Comment by Kvarkas** [comment]
still working

**Comment by Neustradamus** [comment]
@Kvarkas: Badly, already 2 years and 5 months since this ticket.

Latest stable version is 1.76.20 (2019-04-12), no stable version since 6 years, 5 months:
- https://github.com/mRemoteNG/mRemoteNG/releases/tag/v1.76.20

Linked to:
- https://github.com/mRemoteNG/mRemoteNG/issues/1765
- https://github.com/mRemoteNG/mRemoteNG/issues/2028
- https://github.com/mRemoteNG/mRemoteNG/issues/2271
- https://github.com/mRemoteNG/mRemoteNG/issues/2441
- https://github.com/mRemoteNG/mRemoteNG/issues/2452
- https://github.com/mRemoteNG/mRemoteNG/issues/2454
- https://github.com/mRemoteNG/mRemoteNG/issues/2458
- https://github.com/mRemoteNG/mRemoteNG/issues/2474
- https://github.com/mRemoteNG/mRemoteNG/issues/2489
- https://github.com/mRemoteNG/mRemoteNG/issues/2516
- https://github.com/mRemoteNG/mRemoteNG/issues/2552
- https://github.com/mRemoteNG/mRemoteNG/issues/2610

**Comment by Kvarkas** [comment]
do you want to give money so I could hire someone to fix a code or what?

**Comment by mi3technosouth** [comment]
> do you want to give money so I could hire someone to fix a code or what?
@Kvarkas 
Sure. How much we talking big screen tv or US hospital bill?

**Comment by Neustradamus** [comment]
If you do not arrive to fix it, you can search a guy to solve it.

All users would like to be in security.

**Comment by Kvarkas** [comment]
> If you do not arrive to fix it, you can search a guy to solve it.
> 
> All users would like to be in security.

Exactly, but to adress security concerns we shouldnt be in hurry to release but fully test it first from any aspects before release and that takes time as I dont have dedicated testers team and working alone on this project

**Comment by mi3technosouth** [comment]
> > do you want to give money so I could hire someone to fix a code or what?
> > [@Kvarkas](https://github.com/Kvarkas)
> > Sure. How much we talking big screen tv or US hospital bill?

Your website contribute page is not loading anything. How do you want to proceed with the donation?

**Comment by mi3technosouth** [comment]
Would make sense to fork the project and have version that strips the ability to save credentials? Not sure about everyone else but in my personal and business use cases the servers refuse saved credentials and force the user to re-enter in the creds. Therefore I never use the save credentials section.

**Comment by Kvarkas** [comment]
> Your website contribute page is not loading anything. How do you want to proceed with the donation?

working on what, as temporary solution you welcome to check older page version: https://web.archive.org/web/20250122200623/https://mremoteng.org/contribute

**Comment by Kvarkas** [comment]
> Would make sense to fork the project and have version that strips the ability to save credentials? Not sure about everyone else but in my personal and business use cases the servers refuse saved credentials and force the user to re-enter in the creds. Therefore I never use the save credentials section.

technicaly it could be control via settings, so you could restrict saving credentials for all connection (later with implementation of connection profiles, you could set individualy per endpoint)

**Comment by felsgaertner** [comment]
> Would make sense to fork the project and have version that strips the ability to save credentials? Not sure about everyone else but in my personal and business use cases the servers refuse saved credentials and force the user to re-enter in the creds. Therefore I never use the save credentials section.

@mi3technosouth Forking an open source project is quite common in the development community - so don't ask, just do it. Please create a fork and remove that functionality. 

(Perhaps you are the lucky guy with a bigger development team to fix issues faster.)

**Comment by Neustradamus** [comment]
@robertpopa22: No, the latest is always 1.76.20 (2019-04-12).
A lot of CVEs, etc.
